src/psyclone/tests/psyir/transformations/intrinsics/matmul2code_trans_test.py

```diff
@@ -507,64 +531,70 @@ def test_validate11():
     matrix.children[2] = my_range
     with pytest.raises(TransformationError) as excinfo:
         trans.validate(matmul)
-    assert ("To use matmul2code_trans on matmul, only the first two indices "
-            "of the 1st argument are permitted to be Ranges but "
-            "found Range at index 2." in str(excinfo.value))
+    assert ("Transformation Error: To use matmul2code_trans on matmul, "
+            "each Range index of the argument 'x' must be a full range "
+            "but found non full range at position 2." in str(excinfo.value))
```

Comment:
Again, we can use `debug_string()` to improve this error message now.

Reply:
Addressed in 10d5abbfb758dbaa45439829f974a99504e27227.